client/components/foldable-card/index.jsx

```diff
@@ -72,86 +91,79 @@ class FoldableCard extends Component {
-	}
-
-	renderHeader() {
-		const summary = this.props.summary ? (
-			<span className="foldable-card__summary">{ this.props.summary } </span>
-		) : null;
-		const expandedSummary = this.props.expandedSummary ? (
-			<span className="foldable-card__summary-expanded">{ this.props.expandedSummary } </span>
-		) : null;
-		const headerClickAction = this.props.clickableHeader ? this.getClickAction() : null;
-		const headerClasses = classNames( 'foldable-card__header', {
-			'is-clickable': !! this.props.clickableHeader,
-			'has-border': !! this.props.summary,
-		} );
-		return (
-			<div className={ headerClasses } onClick={ headerClickAction }>
-				<span className="foldable-card__main">{ this.props.header } </span>
-				<span className="foldable-card__secondary">
-					{ summary }
-					{ expandedSummary }
-					{ this.renderActionButton() }
-				</span>
-			</div>
-		);
-	}
-
 	render() {
-		const Container = this.props.compact ? CompactCard : Card;
-		const itemSiteClasses = classNames( 'foldable-card', this.props.className, {
-			'is-disabled': !! this.props.disabled,
+		const {
+			actionButton,
+			actionButtonExpanded,
+			className,
+			clickableHeader,
+			compact,
+			disabled,
+			expandedSummary,
+			header,
+			icon,
+			screenReaderText,
+			summary,
+			translate,
+		} = this.props;
+
+		const Container = compact ? CompactCard : Card;
+
+		const itemSiteClasses = classNames( 'foldable-card', className, {
+			'is-disabled': !! disabled,
 			'is-expanded': !! this.state.expanded,
-			'has-expanded-summary': !! this.props.expandedSummary,
+			'has-expanded-summary': !! expandedSummary,
+		} );
+
+		const headerClasses = classNames( 'foldable-card__header', {
+			'is-clickable': !! clickableHeader,
+			'has-border': !! summary,
 		} );
 
 		return (
 			<Container className={ itemSiteClasses }>
-				{ this.renderHeader() }
-				{ this.state.expanded && this.renderContent() }
+				<div
+					className={ headerClasses }
+					onKeyPress={ this.onHeaderKeyPress }
+					onClick={ this.clickHeader }
+					role="button"
+					tabIndex={ 0 }
+				>
+					<span className="foldable-card__main">{ header } </span>
+					<span className="foldable-card__secondary">
+						{ summary && <span className="foldable-card__summary">{ summary } </span> }
+						{ expandedSummary && (
+							<span className="foldable-card__summary-expanded">{ expandedSummary } </span>
+						) }
+						{ actionButton ? (
+							<div
+								className="foldable-card__action"
+								onKeyPress={ this.onActionKeyPress }
+								onClick={ this.clickAction }
+								role="button"
+								tabIndex={ -1 }
+							>
+								{ ( this.state.expanded && actionButtonExpanded ) || actionButton }
+							</div>
+						) : (
+							this.props.children && (
+								<button
+									disabled={ disabled }
+									type="button"
+									className="foldable-card__action foldable-card__expand"
+									onClick={ this.clickExpander }
+									tabIndex={ -1 }
```

Comment:
Same negative tabindex comment. What error were you seeing that a linter wanted you to put a tabIndex on a button? That's not how the `jsx-a11y` rules should work, so if there's a problem with the rules we should investigate.